contributed/Derive_A_EBV_coefficients.ipynb

```diff
@@ -0,0 +1,165 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import GCRCatalogs\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "import scipy.interpolate\n",
+    "from lsst.sims import photUtils\n",
+    "from lsst.sims.photUtils import BandpassSet"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This function grabs the \"LSST_THROUGHPUTS_BASELINE\" filters sa currently specified in the photUtils package, then computes both lambda_eff and A_lambda/E(B-V) for each of the six LSST filters:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def compute_alambda_over_ebv(filterset=['u','g','r','i','z','y'],basepath = None):\n",
+    "    \"\"\"\n",
+    "    compute the effective wavelengths and alambda/E(B-V) values for a set of filters\n",
+    "    We will grab the flat SED from the SIMS library to calculate the CCM dust model \n",
+    "    that was assumed for DC2, and then grab the baseline ugrizy filters and calculate \n",
+    "    their effective wavelengths, and evaluate the CCM alam_over_ebv value at those\n",
+    "    wavelengths\n",
+    "    inputs: filterset:\n",
+    "      vector of filters (limited to ugrizy present for the baseline LSST filterset)\n",
+    "    returns:\n",
+    "    lam_eff_list: \n",
+    "      np 1d array of filter effective wavelengths for the filters\n",
+    "    alam_over_ebv_list:\n",
+    "      np 1d array of alam_over_ebv values for the filters\n",
+    "    \"\"\"\n",
+    "    lam_eff_list = []\n",
+    "    alam_over_ebv_list = []\n",
+    "    sed_file = os.path.join(os.environ['SIMS_SED_LIBRARY_DIR'],'flatSED','sed_flat.txt.gz')\n",
+    "    sed = photUtils.Sed()\n",
+    "    sed.readSED_flambda(sed_file)\n",
+    "    ax,bx = sed.setupCCM_ab()\n",
+    "    ccm_model = 3.1*ax+bx\n",
+    "    wl = sed.wavelen\n",
+    "    ccm_spline = scipy.interpolate.interp1d(wl,ccm_model,bounds_error=True)\n",
+    "    alam_over_ebv = 3.1*ax+bx\n",
+    "    filterset = ['u','g','r','i','z','y']\n",
+    "    for filter in filterset:\n",
+    "        if basepath is None:\n",
+    "            bp_file = os.path.join(os.environ['LSST_THROUGHPUTS_BASELINE'],'',f'total_{filter}.dat')\n",
+    "        else:\n",
+    "            bp_file = os.path.join(basepath,f'total_{filter}.dat')\n",
+    "        bandpass = photUtils.Bandpass()\n",
+    "        bandpass.readThroughput(bp_file)\n",
+    "        _,leff = bandpass.calcEffWavelen()\n",
+    "        lam_eff_list.append(leff)\n",
+    "        #print(f\"band {filter} leff1: {leff:.3f}\")\n",
+    "        alam = ccm_spline(leff)\n",
+    "        #print(f\"leff for band {filter}: {alam:.3f}\")\n",
```

Comment:
Avoid overwriting built-in keyword
```suggestion
    "    for filt in filterset:\n",
    "        if basepath is None:\n",
    "            bp_file = os.path.join(os.environ['LSST_THROUGHPUTS_BASELINE'],'',f'total_{filt}.dat')\n",
    "        else:\n",
    "            bp_file = os.path.join(basepath,f'total_{filt}.dat')\n",
    "        bandpass = photUtils.Bandpass()\n",
    "        bandpass.readThroughput(bp_file)\n",
    "        _,leff = bandpass.calcEffWavelen()\n",
    "        lam_eff_list.append(leff)\n",
    "        #print(f\"band {filt} leff1: {leff:.3f}\")\n",
    "        alam = ccm_spline(leff)\n",
    "        #print(f\"leff for band {filt}: {alam:.3f}\")\n",
```

Reply:
renamed filter to filt to avoid the overwrite